packages/react-components/react-table/src/hooks/useMeasureElement.ts

```diff
@@ -0,0 +1,59 @@
+import { canUseDOM } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+/**
+ * This hook does two things:
+ *  - reports width of the element whose ref has been passed as the parameter
+ *  - as the parent resizes, it sets the ref element width to always fill the parent
+ *
+ * It creates a dummy element adjacent to the one we are trying to measure and
+ * uses it to track the available space for the element.
+ *
+ * This approach is also faster than trying to measure the final element (e.g. a table)
+ * which can take much more time to be laid out than an empty div.
+ *
+ * @param el element to measure and report width on
+ * @returns
+ */
+export function useMeasureElement(el: React.RefObject<HTMLElement>) {
+  const [width, setWidth] = useState(0);
+  const container = useRef<HTMLElement | null>(null);
+  const { current: currentContainer } = container;
```

Comment:
This is incorrect: `container.current` during first render will be always `null` as refs are resolved after rendering. You need to consider to use just state based approach (`React.useState()` or use `useCallbackRef()` similarly to `usePositioning()` hook.